mitodl/open-discussions
# Inject 'is_favorite' and 'lists' fields into search results

#### Pre-Flight checklist

#### What are the relevant tickets?
Fixes #2422

#### What's this PR do?
Appends 'is_favorite' and 'lists' fields/values into each search result if appropriate.

#### How should this be manually tested?
- Go to the course search page and run some searches.  Results that have previously been favorited or added to lists should have a filled-in star.
- Favorite some more results, unfavorite old ones, add & remove results from lists - the star on each result should be updated appropriately.
- Search channels for posts, comments, users should work the same as before.
